### ISSUE TITLE
Improve trust error messages

### ISSUE THREAD

**Issue by tonistiigi** [docker/docker#15463]
Error message shown to people trying out trust could be more verbose and point to possible solution.

##### Pulling untrusted image from hub gives:

```
no trust data available
```

Could be:

```
Error: notary.docker.io does not have trust data for repository foo/bar. Refusing to download image.
```

##### Pushing to a private registry(no notary set up):

```
DEBU[0000] open /root/.docker/trust/tuf/registry.local/busybox/metadata/root.json: no such file or directory
DEBU[0000] Repository not initialized during Publish
Enter key passphrase for offline key with id d6c735d:
DEBU[0003] notary Initialize
DEBU[0003] Adding cert with certID: 23c02a51d689cd75c35913835bc07ede77874a070900c4e5a4760add610fef83
DEBU[0003] Linking 23c02a51d689cd75c35913835bc07ede77874a070900c4e5a4760add610fef83 to d6c735d1b682cab57be39a84707db599654d4f31dfd4dd5df6d9906b4fac3261.
DEBU[0003] Notary syntax error: invalid character 'p' after top-level value
no trust data available for remote repository
```

If its non-TLS registry then this(not sure if it makes sense to deal with this at all):

```
Error establishing connection to notary repository: Get https://registry.local/v2/: tls: oversized record received with length 20527
Get https://registry.local/v2/: tls: oversized record received with length 20527
```

Could be something that lets the user know that they need to set up a notary server and possibly use a `DOCKER_CONTENT_TRUST_SERVER` to point to it.

**Comment by diogomonica** [comment]
Completely agreed on both points. We'll try to get PR out to make this better.

**Comment by endophage** [comment]
Yep. We can and should make some improvements for 1.8.2.

**Comment by riyazdf** [comment]
The errors are split between notary and docker, I've opened a PR for notary and will include changes for docker in a preexisting PR https://github.com/docker/docker/pull/18887

**Comment by thaJeztah** [comment]
@riyazdf if the changes can also be done separately; small PR's are generally easier to review, and will be merged faster. Not sure if it's possible in this case, but just mentioning.

**Comment by riyazdf** [comment]
@thaJeztah I've included some changes in #18887 for now, and I figure it might be nice to have a full-view of the error messages with the new notary errors vendored into that PR as well.  If a separate PR makes more sense that's fine too, I'm open to suggestions :)

**Comment by riyazdf** [comment]
Looping back on this -- this should be improved with https://github.com/docker/notary/pull/419 and #18887, feel free to let me know if there are other errors you think should be improved :)
